e2e/wdio/core/tests/date-picker.e2e-spec.ts

```diff
@@ -0,0 +1,447 @@
+        click(activeButtonDatePicker);
+        click(dayInCalendarButtonByValue('1'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model')).toEqual(date);
+        click(activeButtonDatePicker);
+        expect(highlightedColor)
+            .toContain(getCSSPropertyByName(dayInCalendarButtonByValue('1'), 'background-color').value);
+    });
+
+    it('verify selecting a year range navigates back to the year view', () => {
+        const buttonsLength = getElementArrayLength(activeButtonDatePicker);
+        for (let i = 1; i < buttonsLength; i++) {
+            click(activeButtonDatePicker, i);
+            click(buttonSelectYear);
+            click(buttonSelectYearsRange);
+            waitForElDisplayed(filterCalendarValue('aggregated-year'));
+            click(buttonFirstRangeYear);
+            waitForElDisplayed(filterCalendarValue('year'));
+            click(activeButtonDatePicker, i);
+        }
+    });
+
+    it('verify after the user selects a year, the view changes back to the day view. ', () => {
+        const buttonsLength = getElementArrayLength(activeButtonDatePicker);
+        for (let i = 1; i < buttonsLength; i++) {
+            click(activeButtonDatePicker, i);
+            click(buttonSelectYear);
+            waitForElDisplayed(filterCalendarValue('year'));
+            click(buttonFirstYear);
+            waitForElDisplayed(filterCalendarValue('day'));
+            click(activeButtonDatePicker, i);
+        }
+    });
+
+    it('verify after the user selects a month, the view changes back to the day view. ', () => {
+        const buttonsLength = getElementArrayLength(activeButtonDatePicker);
+        for (let i = 1; i < buttonsLength; i++) {
+            click(activeButtonDatePicker, i);
+            click(buttonSelectMonth);
+            waitForElDisplayed(filterCalendarValue('month'));
+            click(buttonFirstMonth);
+            waitForElDisplayed(filterCalendarValue('day'));
+            click(activeButtonDatePicker, i);
+        }
+    });
+
+    it('verify user is not able select multiple dates', () => {
+        click(activeButtonDatePicker);
+        click(dayInCalendarButtonByValue('1'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model')).toEqual('3/1/2021');
+        click(activeButtonDatePicker);
+        click(dayInCalendarButtonByValue('2'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model')).toEqual('3/2/2021');
+    });
+
+    it('verify disabled date-picker pickers', () => {
+        const disabledDivDatePickerLength = getElementArrayLength(disabledDivDatePicker);
+        for (let i = 0; i < disabledDivDatePickerLength; i++) {
+            scrollIntoView(disabledDivDatePicker, i);
+            expect(getAttributeByName(disabledDivDatePicker, 'class')).toContain('is-disabled');
+        }
+    });
+
+    describe('Check visual regression', function() {
```

Comment:
If you have this kind of repetitive check for each button you can use something similar to 

`    function checkElementStates(selector: string, tag: string, elementName: string, index: number = 0): void {
        checkHoverState(selector, tag + '-hover', elementName, index);
        checkFocusState(selector, tag + '-focus', elementName, index);
        checkActiveState(selector, tag + '-active', elementName, index);
    }`

to have one it for each button state + less tags name needs to be hardcoded in the tags fixture.

P.S. This is an example from select component for core.